test/test_torch.py

```diff
@@ -6892,6 +6893,86 @@ def test_inverse(self, device):
         torch.inverse(matrix, out=matrix_inverse_out)
         self.assertEqual(matrix_inverse_out, matrix_inverse, 0, 'inverse value in-place')
 
+    def test_warning_catch(self):
+        source = '''
+        at::Tensor uncatched(at::Tensor x) {
+            TORCH_WARN("foo");
+            return x.cos();
+        }
+
+        // error_type:
+        // 0: no error
+        // 1: torch::TypeError
+        // 2: python_error()
+        // 3: py::error_already_set
+        at::Tensor catched(at::Tensor x, int error_type) {
+            HANDLE_TH_ERRORS
```

Comment:
Heh, we call it `HANDLE_TH_ERRORS` in our codebase because legacy mumble; no reason to call it that here. `TORCH_HANDLE_ERRORS` maybe?

Reply:
As described in details in the doc, the new macro is not superseeding the old one, it is just used in a different context.
I would be in favor of keeping the same name base here to specify that it is just a specialization of the original macro for pybind-bound functions.
But if you really think we should change, I can do it.